docs/reference/projectyml.rst

```diff
@@ -50,66 +50,165 @@ The code snippet below provides an example of a complete ``project.yml`` file:
+
+Setting this to ``true`` will have the following consequences:
+
+- If no index is set in the project.yml file i.e. both ``index-url`` and ``extra-index-url`` are unset, then Pip's
+  default search behaviour will be used: environment variables, pip config files and then PyPi (in that order).
+
+- If ``index-url`` is set, this value be used over any index defined in the system's environment
+  variables or pip config files.
+
+- If ``extra-index-url`` is set, these indexes will be used in addition to any extra index defined in the system's
+  environment variables or pip config files, and passed to pip as extra indexes.
+
+- If ``PIP_PRE`` is set, pre-release
+  versions are allowed when installing v2 modules or v1 modules' dependencies.
+
+- Each component of the orchestrator, including the agents, will respect their system config, therefore each of these
+  components need to be configured correctly. Auto-started agents live on the same host as the server, so they can
+  easily share config at the system level.
+
+
+.. warning::
+
+    ``use-system-config = true`` should only be used if the pip configuration is fully managed at the system level
+    and secure for each component of the orchestrator.
+
+Example scenario
+""""""""""""""""
+
+1) During development
+
+Using a single pip index isolated from any system config is the recommended approach. The ``pre=true`` option allows
+pip to use pre-release versions, e.g. when testing dev versions of modules published to the dev index. Here is an
+example of a dev config:
 
 .. code-block:: yaml
 
     pip:
-      use_config_file: True
+        index-url: https://my_repo.secure.company.com/repository/dev
+        extra-index-url: []
+        pre: true
+        use-system-config: false
+
+2) In production
 
-To specify the url of a pip repository, add the following to the pip config file of the ``inmanta`` user, located at ``/var/lib/inmanta/.config/pip/pip.conf``:
+Using a single pip index is still the recommended approach, and the use of pre-release versions should be disabled.
 
-.. code-block:: text
 
-  [global]
-  timeout = 60
-  index-url = <url of a python package repository>
+If the pip configuration is fully managed at the system level and secure, use ``use-system-config: true``
+and add any potential project-specific repo that is not managed at the system level through the ``extra-index-url``
+option e.g.:
 
+.. code-block:: yaml
 
-Alternatively, a pip config file can be used at a custom location.
-The ``index-url`` can be specified in this file as explained in the previous section.
-To make this work, the ``PIP_CONFIG_FILE`` environment variable needs to be set to the path of the newly created file (See: :ref:`env_vars`).
-For more information see the `Pip documentation <https://pip.pypa.io/en/stable/topics/configuration/>`_.
+    pip:
+        extra-index-url: [https://my_repo.secure.company.com/repository/project-alpha-production]
```

Comment:
I would not use the words `alpha` and `production` in the same  url: `alpha` often means not ready for production